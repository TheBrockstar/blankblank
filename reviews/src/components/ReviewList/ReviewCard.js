import React from 'react';

const ReviewCard = props => {
  return (
  <article>
    <h1>Author: { props.review.author }</h1>
    <p>{ props.review.body }</p>
  </article>
  )
}

export default ReviewCard