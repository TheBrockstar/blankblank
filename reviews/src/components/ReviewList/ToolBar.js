import React from 'react';
import { ToolBarWrapper, ToolBarContainer } from './Styles';

const ToolBar = props => {
  return (

      <ToolBarWrapper>

      </ToolBarWrapper>

  )
}

export default ToolBar