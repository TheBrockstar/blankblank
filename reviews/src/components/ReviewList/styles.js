import styled from 'styled-components';
import { flexContainer } from '../../globalStyles/Styles';

export const Card = styled.article`
width: 48%;
display: flex;
justify-content: flex-start;
margin-top: 15px;
padding: 15px 10px 15px 15px;
border-radius: 5px;
box-shadow: 0px 1px 2px 1px rgba(126, 56, 12, .5);
background-color: rgba(254, 197, 107, 0.3);

@media only screen and (max-width: 850px) {
  width: 96%;
}
`;

export const CardContent = styled(flexContainer)`
width: 70%;
flex-direction: column;
align-items: flex-start;

@media only screen and (max-width: 450px) {
  width: 65%;
}


h1 {
  width: 100%;
  font-size: 1.6rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: 2.8rem;
}

p {
  width: 100%;
  font-size: 1rem;
  white-space: nowrap;
  overflow-x: hidden;
  text-overflow: ellipsis;
  line-height: 1.5rem;
}
`

export const CardRating = styled(flexContainer)`
width: 100px;
flex-direction: column;
font-size: 1.2rem;
background-color: whitesmoke;
border-radius: 7px;
background-color: rgba(53, 23, 6, 0.5);
color: #51c5d4;
font-weight: bold;
margin-right: 15px;

span {
  font-size: 3.2rem;
}


@media only screen and (max-width: 450px) {
  width: 80px;
  span {
    font-size: 2.3rem;
  }
}
`

export const ListContainer = styled.section`
width: 100%;
display: flex;
justify-content: center;
`;

export const ListWrapper = styled(flexContainer)`
width: 100%;
max-width: 1080px;
justify-content: space-around;
flex-wrap: wrap;

@media only screen and (max-width: 850px) {
  flex-direction: column;
}
`

export const outerStars = styled.div`

`

export const innerStars = styled.div`

`