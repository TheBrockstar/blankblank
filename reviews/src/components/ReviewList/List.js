import React, { useState, useEffect } from 'react';
import axios from 'axios';

import { searchReviews, sortReviews, paginateReviews } from '../../helpers';

import { ListWrapper } from './Styles';
import { Loading, NotFound404 } from '../../globalStyles/Styles'
import ReviewCard from './ReviewCard';

function List(props) {

  const [reviews, setReviews] = useState(null);

  useEffect(() => {
  async function fetchReviews() {
    try {
      const response = await axios(
      'https://shakespeare.podium.com/api/reviews', 
      { 'headers': { 'x-api-key':'H3TM28wjL8R4#HTnqk?c' }}
      );
        
      props.setSize(Math.ceil((response.data.length) / 20));
      setReviews(response.data);
    } catch (error) {
      setReviews('404');
    }
  }
  
  fetchReviews();
  }, []);

  if (reviews) {
    return reviews === '404' ? (<NotFound404>404 - Not Found <br/> Get Thee Hence!</NotFound404>) : (
      <ListWrapper className="listWrapper" role='list'>
        {paginateReviews(sortReviews(searchReviews(reviews, props.searchTerm), props.sortBy, props.sortOrder), props.page, 20).map(review => {
          return (<ReviewCard history={props.history} key={review.id} review={review}/>)
        })}
      </ListWrapper>
    )
  } else {
    return (<Loading>Loading...</Loading>)
  }

}

export default List;

