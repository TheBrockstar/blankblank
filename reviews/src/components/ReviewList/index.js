import React, { useState, useEffect } from 'react';
import axios from 'axios';

import List from './List';

function ReviewList() {

  return (
    <section>
      <List />
    </section>
  )
}

export default ReviewList;

