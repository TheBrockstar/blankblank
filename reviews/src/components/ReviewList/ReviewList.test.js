import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ReviewList from './index';

describe ('ReviewList', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ReviewList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    expect(shallow(<ReviewList />)).toMatchSnapshot();
  });

});