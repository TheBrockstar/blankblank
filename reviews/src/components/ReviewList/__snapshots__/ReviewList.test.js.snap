// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading...",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
